pkg/streamingpromql/operators/functions/range_vectors.go

```diff
@@ -375,3 +375,55 @@ func changes(step *types.RangeVectorStepData, _ float64, _ types.EmitAnnotationF
 
 	return changes, true, nil, nil
 }
+
+var Resets = FunctionOverRangeVectorDefinition{
+	SeriesMetadataFunction: DropSeriesName,
+	StepFunc:               resets,
+}
+
+func resets(step *types.RangeVectorStepData, _ float64, _ types.EmitAnnotationFunc) (float64, bool, *histogram.FloatHistogram, error) {
+	fHead, fTail := step.Floats.UnsafePoints()
+	hHead, hTail := step.Histograms.UnsafePoints()
+
+	haveFloats := len(fHead) > 0 || len(fTail) > 0
+	haveHistograms := len(hHead) > 0 || len(hTail) > 0
```

Comment:
[nit] The head slice will always be populated if there are any points, so we could drop the `len(xTail)` checks.

Reply:
Copied from `xOverTime` function 😆 . Such [as](https://github.com/grafana/mimir/blob/03fc64b8ab4b3230e1ac12344b1733a08453f995/pkg/streamingpromql/operators/functions/range_vectors.go#L139). So this means, we can drop tail check there too, right? Maybe we can do it in another PR.

Reply:
Yep, we can drop it there too, and I'm happy for you to do that in another PR.